src/main/java/net/consensys/orion/storage/QueryPrivacyGroupStorage.java

```diff
@@ -89,4 +83,19 @@ public String generateDigest(QueryPrivacyGroupPayload data) {
       return Optional.of(queryPrivacyGroupPayload);
     });
   }
+
+  private QueryPrivacyGroupPayload handleAlreadyPresentUpdate(
+      QueryPrivacyGroupPayload data,
+      QueryPrivacyGroupPayload result) {
+    List<String> listPrivacyGroupIds;
+    QueryPrivacyGroupPayload queryPrivacyGroupPayload;
+    if (data.isToDelete()) {
+      result.privacyGroupId().remove(data.privacyGroupToAppend());
```

Comment:
This is a bit of a smell - we're deleting the thing flagged as "to be appended".
Wondering if QueryPrivacyGroupPayload needs to be split into Add and Remove classes?
Bare minimum might be to rename "privacyGroupToAppend to "privacyGroupToModify" (or something otherwise better).

Reply:
This smell isn't actually new code in this PR, but I agree & I've renamed accordingly.